tests/cypress/integration/actions_users/issue_1810_login_logout.js

```diff
@@ -96,5 +96,37 @@ context('When clicking on the Logout button, get the user session closed.', () =
             cy.url().should('include', '/auth/login');
             cy.closeNotification('.cvat-notification-notice-login-failed');
         });
+
+        it('Login with Google and GitHub. Logout', () => {
+            let socialAuthMethods;
+            cy.request({
+                method: 'GET',
+                url: '/api/auth/social/methods/',
+            }).then((response) => {
+                socialAuthMethods = Object.keys(response.body);
+                expect(socialAuthMethods).length.gt(0);
+                cy.visit('auth/login');
+
+                cy.get('.cvat-social-authentication-icon').should('have.length', socialAuthMethods.length).within((items) => {
+                    for (const item of items) {
+                        expect(item.children.length).to.be.equal(1); // check that icon was received from the server
+                    }
+                });
+
+                for (const provider of socialAuthMethods) {
+                    let username = '';
+                    cy.get(`.cvat-social-authentication-${provider}`).should('be.visible').click();
+                    // eslint-disable-next-line cypress/no-unnecessary-waiting
+                    cy.wait(3000);
+                    cy.get('.cvat-right-header').within(() => {
```

Comment:
Can we try istead of ``wait()`` something like this?

```suggestion
                    cy.get('.cvat-right-header').should('exist').and('be.visible').within(() => {
```

Cypress itself will wait for element to exist if we require it (up to timeout value)